FileSpecs/Events.md

```diff
@@ -0,0 +1,265 @@
+|               |       | `0a` | U0a.1 | U0a.2 | `00`  | U0a.4 | `00`   | `00`  | `00`    | `00`  | `00`  |
+| **RestoreLP** | 2D    | `0b` | `00`  | `00`  | `00`  | *msg* | `00`   | `00`  | `00`    | `00`  | `00`  |
+| **RestoreSP** | 2D    | `0c` | `00`  | `00`  | `00`  | *msg* | `00`   | `00`  | `00`    | `00`  | `00`  |
+|               |       | `0d` | U0d.1 | `00`  | U0d.3 | `1a`  | U0d.5  | `00`  | `00`    | `00`  | `00`  |
+|               |       | `0e` | U0e.1 | U0e.2 | U0e.3 | U0e.4 | U0e.5  | U0e.6 | U0e.7   | U0e.8 | `00`  |
+|               |       | `0f` | U0f.1 | `01`  | U0f.3 | U0f.4 | U0f.5  | `00`  | `00`    | `00`  | U0f.9 |
+|               |       | `10` | U10.1 | U10.2 | `00`  | U10.4 | U10.5  | `00`  | U10.7   | `00`  | `00`  |
+|               |       | `11` | U11.1 | `00`  | U11.3 | U11.4 | U11.5  | `0`   | U11.7   | `00`  | `00`  |
+|               |       | `12` | U12.1 | U12.2 | U12.3 | U12.4 | `00`   | `00`  | U12.7   | `00`  | U12.9 |
+|               |       | `13` | U13.1 | U13.2 | `00`  | U13.4 | U13.5  | U13.6 | U13.7   | `00`  | U13.9 |
+| **Door3D**    | 3D    | `14` | `01`  | `00`  | `00`  | U14.4 | U14.5  | `00`  | U14.7   | `00`  | `00`  |
+|               | 2D    | `15` | U15.1 | U15.2 | U15.3 | `00`  | `00`   | `00`  | U15.7   | `00`  | `00`  |
+|               | 2D    | `16` | `00`  | `00`  | `00`  | `00`  | `00`   | `00`  | `00`    | `00`  | `00`  |
+| **Win**       | 2D    | `17` | `00`  | `00`  | `00`  | `00`  | `00`   | `00`  | `00`    | `00`  | `00`  |
+
+
+* Column `Where` indicates whether the event only appears in 2D or 3D maps.
+* Parameters of the form "U00.0" indicate an unknown parameter.
+* `00` indicates that this paramter always appears as `00`.
+
+This information is observational: the Amberstar *engine* might support more options values, but the *game* does not seeem to use them (unless events can be triggered in other ways than observed so far).
+
+<!-- ---------------------------------------- -->
+## Event 01: Teleport
+
+Teleports the party to the specified *map* - 1, coordinates (*x*, *y*).
+
+Unknown: *map*=0 might teleport the player to a different location in the same map (?)
+
+### Unknowns
+* U01.3: [`00-03`]
+
+<!-- ---------------------------------------- -->
+## Event 02: Door
+
+Party encounters a locked door.  The lock-picking difficulty seems to be indicated by *cr*, which is 1, 10, or 100.
+
+### Unknowns
+* Exact meaning of the challenge rating *cr*
+* U02.2: [`00`, `01`, `03` (only 3D maps)]
+* U02.3: [`00`, `0a`], on 3D maps also possibly [`05`, `0f`, `23`]
+* U02.5: [`01-0f`], but not `0c-0d`.
+* U02.6: `00`, but may be `01` on 3D maps
+* U02.7: `00` or special number, likely the key ID?  Is nonzero iff *cr* = 100
+
+<!-- ---------------------------------------- -->
+## Event 03: Message
+*Trigger*: Entering, if U03.3 is 0, or Looking if U03.3 is 1.  (Other values for U03.3: Unknown.)
```

Comment:
In Ambermoon this is similar. Bit 0 allows triggering through movement and bit 1 allows triggering by looking. So the value 3 would mean both is possible. I guess a value of 0 would also default to that and maybe just skips the "trigger" check.